[jsp] Parse error on HTML boolean attribute

**Description:**
When parsing a Jsp containing HTML which contains [boolean attributes](http://w3c.github.io/html/infrastructure.html#sec-boolean-attributes), the following error message is produced:
```java
Caused by: net.sourceforge.pmd.lang.jsp.ast.ParseException: Encountered " <ATTR_NAME> "name "" at line 8, column 39.
Was expecting:
    "=" ...
```

**Code Sample demonstrating the issue:**
In the below, the attribute `checked` causes a parse error
```html
<label><input type='checkbox' checked name=cheese disabled=""> Cheese</label>
```

**Running PMD through:** *Ant*

--- a/pmd-jsp/etc/grammar/JspParser.jjt
+++ b/pmd-jsp/etc/grammar/JspParser.jjt
@@ -1,4 +1,8 @@
-/* 
+/*
+ * Allow boolean attributes
+ *
+ * Juan Martín Sotuyo Dodero 06/2017
+ *====================================================================
  * Added capability for Tracking Tokens.
  *
  * Amit Kumar Prasad 10/2015
@@ -580,7 +584,7 @@ void Attribute() :
   (
   	<ATTR_EQ> 
   	AttributeValue()
-  )
+  )?
 }
 
 /**

--- a/pmd-jsp/src/test/java/net/sourceforge/pmd/lang/jsp/JspParserTest.java
+++ b/pmd-jsp/src/test/java/net/sourceforge/pmd/lang/jsp/JspParserTest.java
@@ -29,6 +29,16 @@ public class JspParserTest {
                 "<span class=\"CostUnit\">$</span><span class=\"CostMain\">129</span><span class=\"CostFrac\">.00</span>");
         Assert.assertNotNull(node);
     }
+    
+    /**
+     * Verifies bug #311 Jsp parser fails on boolean attribute
+     */
+    @Test
+    public void testParseBooleanAttribute() {
+        Node node = parse(
+                "<label><input type='checkbox' checked name=cheese disabled=''> Cheese</label>");
+        Assert.assertNotNull(node);
+    }
 
     private Node parse(String code) {
         LanguageVersionHandler jspLang = LanguageRegistry.getLanguage(JspLanguageModule.NAME).getDefaultVersion()

--- a/src/site/markdown/overview/changelog.md
+++ b/src/site/markdown/overview/changelog.md
@@ -63,6 +63,8 @@ Fields using generics are still Work in Progress, but we expect to fully support
     *   [#428](https://github.com/pmd/pmd/issues/428): \[java] PMD requires public modifier on JUnit 5 test
 *   java-unnecessary
     *   [#421](https://github.com/pmd/pmd/issues/421): \[java] UnnecessaryFinalModifier final in private method
+*   jsp
+    *   [#311](https://github.com/pmd/pmd/issues/311): \[jsp] Parse error on HTML boolean attribute
 
 ### API Changes
 

